src/GraphQL/Types/Scalars/EnumerationGraphType.cs

```diff
@@ -62,39 +62,59 @@ public void AddValue(EnumValueDefinition value)
         public EnumValues Values { get; }
 
         /// <inheritdoc/>
-        public override object Serialize(object value)
+        public override object ParseLiteral(IValue value) => value switch
         {
-            var valueString = value.ToString(); //TODO: find only by value?
-            var foundByName = Values.FindByName(valueString);
-            if (foundByName != null)
-            {
-                return foundByName.Name;
-            }
+            EnumValue enumValue => Values.FindByName(enumValue.Name)?.Value ?? ThrowLiteralConversionError(value),
+            NullValue _ => null,
+            _ => ThrowLiteralConversionError(value)
+        };
 
-            var foundByValue = Values.FindByValue(value);
-            return foundByValue?.Name;
-        }
+        /// <inheritdoc/>
+        public override bool CanParseLiteral(IValue value) => value switch
+        {
+            EnumValue enumValue => Values.FindByName(enumValue.Name) != null,
+            NullValue _ => true,
+            _ => false
+        };
+
+        /// <inheritdoc/>
+        public override object ParseValue(object value) => value switch
+        {
+            string s => Values.FindByName(s)?.Value ?? ThrowValueConversionError(value),
+            null => null,
+            _ => ThrowValueConversionError(value)
+        };
```

Comment:
This is what's needed to also parse internal values:

```suggestion
        public override object ParseValue(object value)
        {
            if (value is string s)
            {
                //search externally represented values
                var match2 = Values.FindByName(s);
                if (match2 != null)
                    return match2.Value;
            }

            if (value == null)
                return null;

            //search internally represented values
            var match = Values.FindByValue(value);
            if (match != null)
                return match.Value;

            return ThrowValueConversionError(value);
        }
```

Reply:
Hmm. maybe move `f (value == null)            return null;` after `FindByValue` block right before ThrowValueConversionError ?

Reply:
Then enumeration will support `null` as internal value.

Reply:
🤦 No. `FindByValue(null)` yields match which `Value` will be `null` :)

Reply:
So... I suggest to move null check at the very beginning.

Reply:
An appropriate test would need to be added as well, of course, like the sample provided above:

```cs
        [Fact]
        public void parses_from_value()
        {
            type.CanParseValue(Colors.Red).ShouldBeTrue();
            type.ParseValue(Colors.Red).ShouldBe(Colors.Red);
        }
```

Reply:
Agree.